cli/internal/backends/nodejs/nodejs.go

```diff
@@ -34,6 +34,35 @@ var NodejsYarnBackend = api.LanguageBackend{
 	GetRunCommand: func() []string {
 		return []string{"yarn", "run"}
 	},
+	FinalCheck: func() bool {
+		return !fs.FileExists(".yarnrc.yml")
+	},
+}
+
+var NodejsBerryBackend = api.LanguageBackend{
+	Name:             "nodejs-berry",
+	Specfile:         "package.json",
+	Lockfile:         "yarn.lock",
+	FilenamePatterns: nodejsPatterns,
+	GetWorkspaceGlobs: func() ([]string, error) {
+		pkg, err := fs.ReadPackageJSON("package.json")
+		if err != nil {
+			return nil, fmt.Errorf("package.json: %w", err)
+		}
+		if len(pkg.Workspaces) == 0 {
+			return nil, fmt.Errorf("package.json: no workspaces found. Turborepo requires Yarn workspaces to be defined in the root package.json")
+		}
+		return pkg.Workspaces, nil
+	},
+	GetPackageDir: func() string {
```

Comment:
I think this is only the case with node modules linker, not sure if / how we want to deal with PnP here